src/gobstuf/lib/communicatie.py

```diff
@@ -0,0 +1,291 @@
+        pass
+
+
+class Persoon():
+
+    def __init__(self, persoonsgegevens):
+        """
+        Attribuut aanhef wordt samengesteld op basis van:
+        - voorvoegselGeslachtsnaam
+        - geslachtsnaam
+        - adellijkeTitel_predikaat
+        - geslachtsaanduiding
+        - aanduidingAanschrijving
+        - voorvoegselGeslachtsnaam partner
+        - geslachtsnaam partner
+        - adellijkeTitel_predikaat partner
+        - geslachtsaanduiding partner
+
+        :param persoonsgegevens:
+        """
+
+        self.voorvoegsel_geslachtsnaam = _get_value(persoonsgegevens, 'naam', 'voorvoegsel') or ''
+        self.voorletters = _get_value(persoonsgegevens, 'naam', 'voorletters') or ''
+        self.geslachtsnaam = _get_value(persoonsgegevens, 'naam', 'geslachtsnaam') or ''
+        self.adellijke_titel_predikaat = None
+        self._geslachtsaanduiding = _get_value(persoonsgegevens, 'geslachtsaanduiding') or ''
+        self._aanduiding_naamgebruik = _get_value(persoonsgegevens, 'naam', 'aanduidingNaamgebruik') or ''
+
+    @property
+    def geslachtsaanduiding(self):
+        """
+        Returns the Geslachtsaanduiding for this person
+        :return:
+        """
+        return Geslachtsaanduiding().identifiers.get(self._geslachtsaanduiding)
+
+    @property
+    def aanduiding_naamgebruik(self):
+        """
+        Returns the AanduidingNaamgebruik for this person
+
+        :return:
+        """
+        return AanduidingNaamgebruik().identifiers.get(self._aanduiding_naamgebruik)
+
+
+class Partner(Persoon):
+
+    def __init__(self, persoonsgegevens_partner):
+        """
+        A Partner is a Person with who a Person can relate,
+        either by a huwelijk or a partnerschap
+
+        A relation has a begin date and an optional end date
+
+        :param persoonsgegevens_partner:
+        """
+        self._aangaan_huwelijk_partnerschap = persoonsgegevens_partner.get('aangaanHuwelijkPartnerschap')
+        self._ontbinding_huwelijk_partnerschap = persoonsgegevens_partner.get('ontbindingHuwelijkPartnerschap')
+        # Partner is a persoon
+        super().__init__(persoonsgegevens_partner)
+
+    @property
+    def aangaan_huwelijk_partnerschap_date(self):
+        """
+        Returns the start date for the relation
+
+        :return:
+        """
+        if self._aangaan_huwelijk_partnerschap:
+            return _datum_to_date(self._aangaan_huwelijk_partnerschap.get('datum'))
+
+    @property
+    def ontbinding_huwelijk_partnerschap_date(self):
+        """
+        Returns the end date for the relation
+
+        :return:
+        """
+        if self._ontbinding_huwelijk_partnerschap and\
+                self._ontbinding_huwelijk_partnerschap.get('indicatieHuwelijkPartnerschapBeeindigd'):
+            return _datum_to_date(self._ontbinding_huwelijk_partnerschap.get('datum'))
+
+
+class Communicatie():
+
+    def __init__(self, persoon, partners=None, partnerhistorie=None):
```

Comment:
Nice lib!

However, I would propose one change: Change the signature of this constructor to

`def __init__(self, persoon: Persoon, partners: List[Partner] = [], partnerhistory: List[Partner] = [])`

This avoids the initialisation and type checking of different objects, but above all it makes the signature clear at first glance. A Persoon object is a Persoon object no matter what, when passing a (list of) dicts, you'll have to visit the Persoon and Partner classes to check the validity of the dicts. Makes the class a bit cleaner and clearer.

Reply:
Use of python typing is indeed more clear, I will change this.
Take care however with default parameters as they are mutable:
Example:
```
class X:
    def __init__(self, a=[]):
        a.append(1)
        self.a = a

x1 = X()
print(x1.a)
x2 = X()
print(x2.a)
```
=>
```
[1]
[1, 1]

```
So default parameters should better be None and initialized afterwards like:

```
class X:
    def __init__(self, a=None):
        a = a or []
        a.append(1)
        self.a = a

x1 = X()
print(x1.a)
x2 = X()
print(x2.a)
```
=>

```
[1]
[1]

```

Reply:
Yes you're right, Python logic..